contracts/holder/src/contract.rs

```diff
@@ -0,0 +1,92 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{
+    to_binary, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult, Coin, BankMsg
+};
+use cw2::set_contract_version;
+
+use crate::msg::{ExecuteMsg, InstantiateMsg, QueryMsg};
+use crate::state::{WITHDRAWER};
+use crate::error::ContractError;
+
+const CONTRACT_NAME: &str = "crates.io:covenant-holder";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    // We cannot deserialize the address without first validating it
+    let withdrawer = msg
+        .withdrawer
+        .map(|human| deps.api.addr_validate(&human))
```

Comment:
for the mapping to be more general maybe lets do something like this?
```suggestion
        .map(|addr| deps.api.addr_validate(&addr))
```